merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/utils/JavadocParser.java

```diff
@@ -0,0 +1,175 @@
+   * @param elementUtils Utility functions for file parsing.
+   * @param element The element we're parsing the comments for.
+   * @return A map of resource types to their associated unit.
+   */
+  public static Map<String, String> parseMissionModelUnits(Elements elementUtils, TypeElement element) {
+    final var parsedUnits = new HashMap<String, String>();
+
+    // Parse the class javadoc.
+    Optional.ofNullable(elementUtils.getDocComment(element))
+        .map(JavadocParser::removeSingleLeadingSpaceFromEachLine)
+        .map(comment -> {
+          parseComment(comment, parsedUnits, RESOURCE_TYPE_TAG);
+
+          return comment;
+        });
+
+    element
+        .getEnclosedElements()
+        .stream()
+        .flatMap(e -> Optional
+            .ofNullable(elementUtils.getDocComment(e))
+            .map(JavadocParser::removeSingleLeadingSpaceFromEachLine)
+            .map(comment -> Pair.of(e.getSimpleName().toString(), comment))
+            .stream())
+        .forEach($ -> {
+          final var value = $.getValue();
+          var property = $.getKey();
+
+          if (StringUtils.containsIgnoreCase(value, UNITS_TAG)) {
+            // Prepend the registered state with a "/" as it's removed from the property name.
+            if (StringUtils.containsIgnoreCase(value, RESOURCE_TYPE_TAG)) {
+              property = "/" + property;
+            }
+
+            parsedUnits.put(property, extractTagValue(value));
+          }
+        });
+
+    return parsedUnits;
+  }
+
+  /**
+   * Parses javadocs searching for parameters + unit paris and computed attributes + unit pairs.
+   * @param elementUtils Utility functions for file parsing.
+   * @param element The element we're parsing the comments for.
+   * @return A pair of maps, one with the parameter units and one with the computed attribute units.
+   */
+  public static Pair<Map<String, String>, Map<String, String>> parseActivityTypeUnits(
+      Elements elementUtils,
+      TypeElement element
+  ) {
+    final var parameterUnits = new HashMap<String, String>();
+    final var computedAttributeUnits = new HashMap<String, String>();
+
+    // Parse the activity type class javadoc.
+    Optional.ofNullable(elementUtils.getDocComment(element))
+          .map(JavadocParser::removeSingleLeadingSpaceFromEachLine)
+          .map(comment -> {
+            parseComment(comment, parameterUnits, PARAM_TAG);
+
+            return comment;
+          });
+
+    // Parse the parameter javadocs inside the class.
+    element
+        .getEnclosedElements()
+        .stream()
+        .flatMap(e -> Optional
+            .ofNullable(elementUtils.getDocComment(e))
+            .map(JavadocParser::removeSingleLeadingSpaceFromEachLine)
+            .map(comment -> Pair.of(e.getSimpleName().toString(), comment))
+            .stream())
+        .forEach($ -> {
+          final var value = $.getValue();
+          var property = $.getKey();
+
+          System.out.println(value);
```

Comment:
Some lingering debug statements - probably a good idea to remove them to avoid cluttering the compile output

Reply:
Good catch, thought I removed everything!